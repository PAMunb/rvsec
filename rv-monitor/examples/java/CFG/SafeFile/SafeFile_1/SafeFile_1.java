package SafeFile_1;

import java.io.File;
import java.io.FileReader;
import java.io.FileWriter;

public class SafeFile_1{

	static FileReader fr = null;

	public static void sub1(){
		System.out.println("begin");

		File file = null;
		try{
			file = File.createTempFile("rvmonitortest1", ".tmp");
			FileWriter fw_1 = new FileWriter(file);
			fw_1.write("testing\n");
			fw_1.close();
		} catch (Exception e){
			System.out.println(e.getMessage());
		}

		try{
			System.out.println("open");
			fr = new FileReader(file);
			rvm.SafeFileRuntimeMonitor.openEvent(fr, Thread.currentThread());
		} catch (Exception e){
			System.out.println(e.getMessage());
		}
		try{
			System.out.println("close");
			fr.close();
			rvm.SafeFileRuntimeMonitor.closeEvent(fr, Thread.currentThread());
		} catch (Exception e){
		}
		System.out.println("end");
	}

	public static void sub2(){
		System.out.println("begin");

		File file = null;
		try{
			file = File.createTempFile("rvmonitortest1", ".tmp");
			FileWriter fw_1 = new FileWriter(file);
			fw_1.write("testing\n");
			fw_1.close();
		} catch (Exception e){
			System.out.println(e.getMessage());
		}

		try{
			System.out.println("open");
			fr = new FileReader(file);
			rvm.SafeFileRuntimeMonitor.openEvent(fr, Thread.currentThread());
		} catch (Exception e){
			System.out.println(e.getMessage());
		}
		System.out.println("end");
	}

	public static void sub3(){
		System.out.println("begin");
		try{
			System.out.println("close");
			fr.close();
			rvm.SafeFileRuntimeMonitor.closeEvent(fr, Thread.currentThread());
		} catch (Exception e){
			System.out.println(e.getMessage());
		}
		System.out.println("end");
	}

	public static void main(String[] args){
		rvm.SafeFileRuntimeMonitor.beginCallEvent(Thread.currentThread());
		rvm.SafeFileRuntimeMonitor.beginCallEvent(Thread.currentThread());
		sub1();
		rvm.SafeFileRuntimeMonitor.endCallEvent(Thread.currentThread());
		rvm.SafeFileRuntimeMonitor.beginCallEvent(Thread.currentThread());
		sub2();
		rvm.SafeFileRuntimeMonitor.endCallEvent(Thread.currentThread());
		rvm.SafeFileRuntimeMonitor.beginCallEvent(Thread.currentThread());
		sub3();
		rvm.SafeFileRuntimeMonitor.endCallEvent(Thread.currentThread());
		rvm.SafeFileRuntimeMonitor.endCallEvent(Thread.currentThread());
	}

}
