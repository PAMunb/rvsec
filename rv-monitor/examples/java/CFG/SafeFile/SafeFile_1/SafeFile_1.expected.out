begin
open
close
end
begin
open
end
improper use of files
improper use of files
begin
close
improper use of files
end
improper use of files
improper use of files
